privacyidea/lib/utils/__init__.py

```diff
@@ -1135,6 +1139,51 @@ def truncate_comma_list(data, max_len):
     return ",".join(data)
 
 
+def generate_charlists_from_pin_policy(policy):
+    """
+    This function uses the pin content policy string to create the character lists
+    for password generation.
+
+    :param policy: The policy that describes the allowed contents of the PIN (see check_pin_policy)
+    :return: Dictionary with keys "base" for the base set of allowed characters and "requirements"
+     which denotes a list of characters from each of which at least one must be contained in the pin.
+    """
+
+    valid_policy_regexp = re.compile(r'^[+-]*[cns]+$|^\[.*\]+$')
```

Comment:
As this is a fixed value, please capitalize

Reply:
done